system/micd.py

```diff
@@ -16,10 +16,8 @@
 def calculate_spl(measurements):
   # https://www.engineeringtoolbox.com/sound-pressure-d_711.html
   sound_pressure = np.sqrt(np.mean(measurements ** 2))  # RMS of amplitudes
-  if sound_pressure > 0:
-    sound_pressure_level = 20 * np.log10(sound_pressure / REFERENCE_SPL)  # dB
-  else:
-    sound_pressure_level = 0
+  sound_pressure_clamped = max(REFERENCE_SPL, sound_pressure) # Make sure measurements being 0 doesn't cause an issue
+  sound_pressure_level = 20 * np.log10(sound_pressure_clamped / REFERENCE_SPL)  # dB
```

Comment:
the spl from this is only used in one place, maybe we should just clip it when publishing? not sure if it makes sense to clip this function

Reply:
actually it does make sense for the sound pressure to be less than the reference, so sound pressure level being negative also is okay. think it's safe to close